modules/tftp/server.rb

```diff
@@ -63,6 +63,23 @@ def delete_file(file)
         logger.debug "TFTP: Skipping a request to delete a file which doesn't exists"
       end
     end
+
+    def delete_host_dir(mac)
+      host_dir = "#{path}/host_config/#{dashed_mac(mac).downcase}"
+      if Dir.exist?(host_dir)
```

Comment:
`Dir.exist?` and then removing is an anti-pattern because it creates a race condition. While it's minor, I'd prefer to avoid it since you're only using it to have a debug statement.

And I realize `delete_file` already has that, but is that a good reason to copy it?

Reply:
I never thought about it being a bad idea to check if a directory exists before doing something with the directory and I did some research on the internet. I agree: Relying on the existence of a directory in a statement before doesn't provide any benefit and it can possibly lead to race conditions, depending on what you do with the directory afterwards.